packages/hardhat-solhint/test/tests.ts

```diff
@@ -36,11 +36,28 @@ describe("Solhint plugin", function () {
       );
     });
 
-    it("should run the check task without throwing an error", async function () {
+    it("should run the check task and exit", async function () {
       const consoleLogStub = sinon.stub(console, "log");
+      const processExitStub = sinon.stub(process, "exit");
       await this.env.run("check");
       assert.isTrue(consoleLogStub.calledOnce);
+      assert.isTrue(processExitStub.calledOnceWith(1));
```

Comment:
Does process exit need to be a spy rather than a stub. Or put another way, in this test will the process exit actually be called? Is that relevant?
Or is it the case that we don't expect any code after the error count conditional so we don't need to actually trigger the `process.exit` call.

Reply:
Lets not worry about this. It does mean you need to grok the context of the test - `process.exit` isn't being called and that has implications for follow on `check` tasks; but I don't think the trade off here in effort is worth it.

Reply:
Would the ask here be to basically split the test case into 2? One that checks whether we print out the report and a separate one that checks if we properly exit when we find errors?

Reply:
I think we leave this, but in an ideal world I think having two test cases:

1. two check tasks, confirm the second one runs if there are no solhint errors
2. two check tasks, confirm the second one doesn't run if there are solhint errors

The stubbing of `process.exit` would fail here; the stub would be called but it wouldn't "exit", it would continue on.

However, I don't think this particular case justifies the effort of figuring out how to safely `process.exit` in a test. What do you think?

Reply:
Yes, that makes sense and I do agree in this particular case. We can deal with it the next time we encounter a need for a similar check.